Samples/iOS-Swift/iOS-Swift/Profiling/BenchmarkingViewController.swift

```diff
@@ -0,0 +1,124 @@
+        highlightButton(sender)
+        guard let value = SentryBenchmarking.stopBenchmark() else {
+            let alert = UIAlertController(title: "Benchmark Error", message: "No benchmark result available.", preferredStyle: .alert)
+            alert.addAction(UIAlertAction(title: "OK", style: .default))
+            self.present(alert, animated: false)
+            print("[iOS-Swift] [debug] [ProfilingViewController] no benchmark result returned")
+            return
+        }
+        valueTextField.isHidden = false
+        valueTextField.text = value
+        print("[iOS-Swift] [debug] [ProfilingViewController] benchmarking results:\n\(value)")
+    }
+    
+    @IBAction func startCPUWork(_ sender: UIButton) {
+        highlightButton(sender)
+        _adjustWorkThreadsToCurrentRequirement()
+    }
+    
+    @IBAction func minWorkThreadCountChanged(_ sender: Any) {
+        _adjustWorkThreadsToCurrentRequirement()
+    }
+    
+    @IBAction func workThreadSliderChanged(_ sender: Any) {
+        _adjustWorkThreadsToCurrentRequirement()
+    }
+    
+    @IBAction func maxWorkThreadCountChanged(_ sender: Any) {
+        _adjustWorkThreadsToCurrentRequirement()
+    }
+    
+    @IBAction func endCPUWork(_ sender: UIButton) {
+        highlightButton(sender)
+        cpuWorkthreads.forEach { $0.cancel() }
+        cpuWorkthreads.removeAll()
+    }
+    
+    @IBAction func minWorkIntervalChanged(_ sender: Any) {
+        _adjustWorkIntervalToCurrentRequirements()
+    }
+    
+    @IBAction func workIntensityChanged(_ sender: UISlider) {
+        _adjustWorkIntervalToCurrentRequirements()
+    }
+    
+    @IBAction func maxWorkIntervalChanged(_ sender: Any) {
+        _adjustWorkIntervalToCurrentRequirements()
+    }
+    
+    @IBAction func bgBrightnessChanged(_ sender: UISlider) {
+        view.backgroundColor = .init(white: CGFloat(sender.value), alpha: 1)
+    }
+        
+    // MARK: UITextFieldDelegate
+    
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        textField.resignFirstResponder()
+    }
+    
+    func _adjustWorkThreadsToCurrentRequirement() {
+        let maxThreads = (maxThreadsTextField.text! as NSString).integerValue
+        let minThreads = (minThreadsTextField.text! as NSString).integerValue
+        let requiredThreads = _projectedRange(factor: workThreadSlider.value, min: minThreads, max: maxThreads)
+        let diff = requiredThreads - cpuWorkthreads.count
+        if diff == 0 {
+            return
+        } else if diff > 0 {
+            for _ in 0 ..< diff {
+                let thread = WorkThread()
+                thread.qualityOfService = .userInteractive
+                cpuWorkthreads.insert(thread)
+                thread.start()
+            }
+        } else {
+            let absDiff = abs(diff)
+            for _ in 0 ..< absDiff {
+                let thread = cpuWorkthreads.removeFirst()
+                thread.cancel()
+            }
+        }
```

Comment:
`l`: nit pick to make early exit more visible

```suggestion
        if diff == 0 {
            return
        }
        
        if diff > 0 {
            for _ in 0 ..< diff {
                let thread = WorkThread()
                thread.qualityOfService = .userInteractive
                cpuWorkthreads.insert(thread)
                thread.start()
            }
        } else {
            let absDiff = abs(diff)
            for _ in 0 ..< absDiff {
                let thread = cpuWorkthreads.removeFirst()
                thread.cancel()
            }
        }
```